docs/devGuide/workflow.md

```diff
@@ -242,3 +242,19 @@ First, follow the instruction to [delete the dependency](#to-delete-a-dependency
 <box type="warning">
 Dependency updates are not trivial, and can be the source of subtle bugs. You should always check the respective dependency changelogs before doing so!
 </box>
+
+### Points for consideration
+There are a few ways to incorporate external packages into MarkBind, each with its pros and cons. The following table shows some of the common trade-offs:
+
+Approach          | Pros                                                                                                                                                                 | Cons                                                                                                                            |
+:-----------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------|
+Installing | <ul><li>Ease of upgrade</li><li>Traceable in `package.json`</li></ul>                                                                                                | <ul><li>May not satisfy custom behavior</li><li>Become vulnerable if the source repo is no longer actively maintained</li></ul>
+Forking           | <ul><li>Relatively easy to upgrade</li><li>Leverage upstream testing procedures</li><li>Benefit others who share the same use cases</li></ul>                                                           | <ul><li>May become out-of-sync with the latest version</li><li>Difficult to maintain it externally in the long run</li></ul>                   
+Patching          | <ul><li>Quick - no need to maintain / publish more npm packages or setup release procedures etc</li><li>Ensure the changes propagate to other dependencies</li><li>Benefits of a monorepo</li></ul> | <ul><li>Difficult to upgrade</li></ul>                              
```

Comment:
Maybe paraphrase the `Benefits of a monorepo` part to `Enjoy the benefits of monorepo`?